pkg/reconciler/status_test.go

```diff
@@ -0,0 +1,31 @@
+package reconciler
+
+import (
+	"context"
+	"testing"
+
+	provider2 "github.com/openshift-pipelines/pipelines-as-code/pkg/provider"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/test/provider"
+	"go.uber.org/zap"
+	zapobserver "go.uber.org/zap/zaptest/observer"
+	"gotest.tools/v3/assert"
+)
+
+func TestCreateStatusWithRetry(t *testing.T) {
+	observer, _ := zapobserver.New(zap.InfoLevel)
+	fakelogger := zap.New(observer).Sugar()
+	vcx := provider.TestProviderImp{}
+
+	err := createStatusWithRetry(context.TODO(), fakelogger, nil, &vcx, nil, nil, provider2.StatusOpts{})
+	assert.NilError(t, err)
+}
+
+func TestCreateStatusWithRetry_ErrorCase(t *testing.T) {
+	observer, _ := zapobserver.New(zap.InfoLevel)
+	fakelogger := zap.New(observer).Sugar()
+	vcx := provider.TestProviderImp{}
+	vcx.CreateStatusErorring = true
+
+	err := createStatusWithRetry(context.TODO(), fakelogger, nil, &vcx, nil, nil, provider2.StatusOpts{})
+	assert.Error(t, err, "failed to report status: you want me to error I error for you")
```

Comment:
is people going to understand that error 🤔 ?

Reply:
updated msg, merging.